### PR TITLE
Handle metric filters in drilldowns

### LINKED ISSUE #6010: Drill through / underlying records action doesn't respect filters in metric definitions
This is on Chrome on our stats instance.

If I have a "common metric" which includes one or more filters in its definition, when I view it in the query builder and then either use the `underlying records` action, or click on a point on a line and use the `view these items` drill-through action, the list of rows that gets returned don't include the filters that are part of the metric's definition.

For example, say I have a metric called `Revenue from Octogenarian Users`, which is defined as `sum of subtotal where purchaser age is between 80 and 89`.  If I'm viewing it as a time series and I click `view these users` from the drill-through menu on one of the line's points, the raw data query that's subsequently run will correctly filter on the date I clicked on, but it won't include the `age is between 80 and 89` filter. This means I'm viewing the wrong segment.

Here's a very similar example:

**Query using a common metric**

screen shot 2017-09-19 at 11 08 20 am

**Result of viewing underlying records (note the filter isn't there)**

screen shot 2017-09-19 at 11 08 35 am

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/6010
Reproduces https://github.com/metabase/metabase/issues/6010

How to test:
- Go to Admin -> Data Model -> Metrics and create a metric with Filter (Total > 150) and Aggregation (Count)
- New -> Question -> Orders -> Visualize
- Summarize -> Remove the default aggregation -> Add metric -> Choose your metric
- Group by Created At (month)
- Click on any dot -> View these Orders
- Make sure the newly constructed query has the filter from the metric applied

Metric definition:
Screenshot 2022-09-23 at 23 04 11

Adding the metric to a question:
Screenshot 2022-09-23 at 23 05 07

Drilldown:
Screenshot 2022-09-23 at 23 04 01

New query with the metric filter:
Screenshot 2022-09-23 at 23 03 43

### CODE REVIEW

File: frontend/test/__support__/sample_database_fixture.json [hunk 1]
```diff
@@ -57,6 +57,37 @@
         "how_is_this_calculated": null,
         "created_at": "2017-06-14T23:32:12.267Z",
         "points_of_interest": null
+      },
+      "2": {
+        "description": "Because we want to know the total I ugess",
+        "table_id": 1,
+        "definition": {
+          "filter": [">", 6, 20],
```
Comment (ranquild) [inline frontend/test/__support__/sample_database_fixture.json:65]: This is the copy of the previous metric but with a `filter` clause.

File: frontend/src/metabase/modes/lib/actions.js [hunk 2]
```diff
@@ -83,30 +83,6 @@ export function distribution(question, column) {
   }
 }
 
-export function toUnderlyingRecords(question) {
```
Comment (ranquild) [inline frontend/src/metabase/modes/lib/actions.js:86]: Inlined in `Question`

File: frontend/src/metabase/modes/components/drill/UnderlyingRecordsDrill.jsx [hunk 3]
```diff
@@ -21,20 +19,6 @@ export default ({ question, clicked }) => {
   // the metric value should be the number of rows that will be displayed
   const count = typeof clicked.value === "number" ? clicked.value : 2;
 
-  // special case for aggregations that include a filter, such as share, count-where, and sum-where
-  let extraFilter = null;
```
Comment (ranquild) [inline frontend/src/metabase/modes/components/drill/UnderlyingRecordsDrill.jsx:25]: This was moved to `Question.drillUnderlyingRecords`

**Review by daltojohnso — APPROVED** [review]
Not super familiar with this area, but works well.

**Pull request merged**